core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -129,6 +129,11 @@
     RestoreSource restoreSource();
 
     /**
+     * Additional metadata on why the shard is unassigned.
```

Comment:
nit picky - is -> was. we may have the shard assigned and initializing  and still have this info. Might be a good idea to add that to the comment too